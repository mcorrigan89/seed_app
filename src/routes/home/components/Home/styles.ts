import { css } from 'emotion';

export namespace styles {
  export const title = css({
    fontSize: 72
  });
}