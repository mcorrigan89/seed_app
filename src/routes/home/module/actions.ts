import { createAction } from '@helpers/action';

export namespace ActionTypes {
  export const UPDATE_TITLE = '[home] UPDATE_TITLE';
}

export const Actions = {
  updateTitle: (title: string) => createAction(ActionTypes.UPDATE_TITLE, title)
};
