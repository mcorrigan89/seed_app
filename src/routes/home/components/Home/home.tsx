import * as React from 'react';
import { Actions } from '@routes/home/module/actions';

export interface Props {
  title: string;
  updateTitle: typeof Actions.updateTitle;
  clearTitle: typeof Actions.clearTitle;
}

interface State {
  title: string;
}

export class HomeComponent extends React.Component<Props, State> {

  state = {
    title: this.props.title
  };

  handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({ title: e.currentTarget.value });
  }

  handleUpdate = () => {
    this.props.updateTitle(this.state.title);
  }

  public render() {
    return (
      <>
        <div data-attr={'test'}>{this.props.title}</div>
        <label>Update Title
          <input type={'text'} name={'title'} value={this.state.title} onChange={this.handleInputChange} />
        </label>
        <button onClick={this.handleUpdate}>Update</button>
      </>
    );
  }
}