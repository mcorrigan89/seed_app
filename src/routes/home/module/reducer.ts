import { RootActions } from '@store/actions';
import { Reducer } from 'redux';
// import reduceReducers from 'reduce-reducers';

import { ActionTypes } from './actions';

export interface HomeState {
  title: string; 
}

const initialState: HomeState = { title: 'Home' };

export const reducer: Reducer<HomeState, RootActions> = (state = initialState, action): HomeState => {
  switch (action.type) {
    case ActionTypes.UPDATE_TITLE:
      return { ...state, title: action.payload };
    case ActionTypes.CLEAR_TITLE:
      return { ...state, title: '' };
    default:
      return state;
  }
};

export const combinedReducers = reducer;
// export const combinedReducers = reduceReducers<HomeState>(reducer);