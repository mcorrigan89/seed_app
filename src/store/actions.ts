import { ActionsUnion } from '@helpers/action';

import { Actions as HomeActions } from '@routes/home/module/actions';

export type RootActions = ActionsUnion<
  typeof HomeActions>;