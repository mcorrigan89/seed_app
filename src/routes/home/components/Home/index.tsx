import { HomeComponent, Props } from '@routes/home/components/Home/home';
export { HomeComponent, Props };