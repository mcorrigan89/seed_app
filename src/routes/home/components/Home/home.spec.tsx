import { setUpTestsWithDOM } from '@tests/setup';
const Enzyme = setUpTestsWithDOM();
import * as React from 'react';
import { Actions } from '../../module/actions';

import { HomeComponent } from '@routes/home/components/Home';

it('HomeComponent', () => {
    const result = Enzyme.mount(<HomeComponent title={'Flerp'} clearTitle={Actions.clearTitle}/>);
    expect(result.props().title).toEqual('Flerp');
    result.setProps({ title: 'Derp'});
    expect(result.props().title).toEqual('Derp');
});