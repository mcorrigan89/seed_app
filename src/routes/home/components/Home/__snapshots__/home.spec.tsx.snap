// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent Should match snapshot 1`] = `
Array [
  <div
    className="css-5zsq25"
    data-attr="test"
  >
    Flerp
  </div>,
  <label>
    Update Title
    <input
      name="title"
      onChange={[Function]}
      type="text"
      value="Flerp"
    />
  </label>,
  <button
    onClick={[Function]}
  >
    Update
  </button>,
]
`;
